### PR TITLE
The recipe of red phosphorus asked for matchboxes and matchbooks

### PR DESCRIPTION
**Pull request by Lalomb0000**
Changed to ask for the matches instead

for convenience and common sense, the important chemical component of the recipe are the matches

#### Summary
Category "Balance"

#### Purpose of change
The recipe of red phosphorus cannot be done with empty matchboxes, and if you unload them, you cannot put them back in

#### Describe the solution
To make the matches themselves the component of the recipe

#### Describe alternatives you've considered
To add a refilling function to the matchboxes and matchbooks

#### Testing

tested
#### Additional context
none

### CODE REVIEW

**Comment by pjf** [comment]
OMG, I want this change to be merged so much, but I also know where the phosphorous is found in safety matches...

Wikipedia text on safety matches, reading: The dangers of white phosphorus in the manufacture of matches led to the development of the "hygienic" or "safety match". The major innovation in its development was the use of red phosphorus, not on the head of the match but instead on a specially designed striking surface

**Comment by liminalRoar** [comment]
So, why don't we add a chance for strike-anywhere matches? They are common enough as well, and have phosphorus built into the matches.

**Comment by Aeiouuu01234** [comment]
Well, I made this change for convenience’s sake, I know that the matches in themselves don't have phosphorus, another alternative is disassembling the strip, but that will need a lot of matchboxes anyway

**Comment by ANickelN** [comment]
Just for reference, there are two type of matchbooks in the game.  One is the traditional fold over that holds I think 24, and the other is one of the little boxes that hold 36.  The little boxes CAN actually be refilled, so if you're wondering what to do with excess matches I always fill these little boxes with them.

**Comment by prilr** [comment]
> I have a feeling that won't fly personally. The recipe we have now is for using the strip, that is why it uses the matchbox, i thought. Maybe im not fully aware of the issue. I am surprised it doesn't work with an empty matchbox and that they can't be refilled.
> 
> I do think that strike anywhere matches would be a great addition. No box needed, although a waterproof container would be best if you want to take them underwater of course.

Are you sure that the strike surface is the material being used? This discussion seems to suggest otherwise: https://github.com/CleverRaven/Cataclysm-DDA/pull/57335#issuecomment-1114294181

In any case, as things stand, the recipe doesn't quite make sense: the moment you unload a matchbook, you can no longer craft phosphorous with it, even though both the matches and the strip are still available - and you can't put the matches back, either, so the change is irreversible.

If the current recipe actually harvests the phosphorous from the striking surface, an alternative could be making the recipe consume the matchbox, whether empty or full.

**Comment by liminalRoar** [comment]
> Are you sure that the strike surface is the material being used? This discussion seems to suggest otherwise: [#57335 (comment)](https://github.com/CleverRaven/Cataclysm-DDA/pull/57335#issuecomment-1114294181)

https://goneoutdoors.com/strikeanywhere-matches-work-4914070.html

this link states the following:

(in reference to 'how do strike anywhere matches work?') "Because the phosphorus is incorporated into the match itself, the need for the strike strip is eliminated. For the match to ignite, it simply needs to be rubbed against any abrasive surface to generate the heat needed to activate the phosphorous reaction."

**Comment by prilr** [comment]
> as far as i can tell, in reference to that comment, it must be referring to the fact strike anywhere matches are common in the US. which is true. but also the other variant is common as well, as they are cheaper to make and buy. Hence why i think both should be in game.

Yeah, looked it up myself, and it seems that strike-anywhere matches have fallen out of popularity in USA, the matchboxes one can loot from houses likely have phosphorous in the strike surfaces.

So I'm guessing the optimal resolution would be:

- current "matchboxes/books" become safety matchboxes/books;
  - red phosphorous recipe involving those consumes the box/book, whether full or empty;

- new "strike-anywhere matches" are introduced, less common than safety, mostly found in sports stores;
  - red phosphorous recipe involving those consumes the matches themselves, box is not important;

Sounds about right?

**Comment by liminalRoar** [comment]
I would be pretty cool with that. i do think they should have a chance to spawn where normal matchbooks do. even if its a 1/10, 1/20th chance comparatively or less even.

But yeah sounds good

**Comment by FayFlightee** [comment]
You weren't actually going to make matchboxes non-reloadable, right? That just... Isn't right. Over where I live, matchboxes are much more common and since they're just, well, boxes for holding matches, I'm not sure why would they not be reloadable.

File: data/json/recipes/recipe_medsandchemicals.json [hunk 1]
```diff
@@ -1125,7 +1125,7 @@
     "time": "10 m",
     "autolearn": true,
     "qualities": [ { "id": "FINE_GRIND", "level": 1 } ],
-    "tools": [ [ [ "matches", 20 ], [ "ref_matches", 20 ] ] ]
+    "components": [ [ [ "match", 20] ] ]
```
Comment (Night-Pryanik) [inline data/json/recipes/recipe_medsandchemicals.json:1128]:
```suggestion
    "components": [ [ [ "match", 20 ] ] ]
```

**Pull request closed without merging**